### PR TITLE
[README] Add section on 🤗 Transformers

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
First of all, thank you for your amazing work on the Whisper project and for open-sourcing the family of pre-trained checkpoints - these are of tremendous benefit to both the open-source/open-science communities!

Whisper was added to Hugging Face Transformers in v4.23.1 in PyTorch and Tensorflow, making it more accessible to the research community. Since then, usage has been steadily increasing, with model downloads now at ~30k / month. Adding a section on the 🤗 Transformers implementation of Whisper to the README would help highlight how one can evaluate Whisper in ~10 lines of code! We're excited to share with you how the community adopt Whisper for ASR research and production!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,43 @@ result = whisper.decode(model, mel, options)
 print(result.text)
 ```
 
+## Use in 🤗 Transformers
+
+Whisper is available in the Hugging Face [Transformers](https://github.com/huggingface/transformers) library from Version 4.23.1, with both PyTorch and TensorFlow implementations. All the official checkpoints can be found on the [Hugging Face Hub](https://huggingface.co/models?search=openai/whisper), alongside [documentation](https://huggingface.co/docs/transformers/model_doc/whisper) and [examples scripts](https://huggingface.co/blog/fine-tune-whisper).
```
Comment (patrickvonplaten) [inline README.md:134]: Maybe might be good to mention how to install transformers:
```suggestion
Whisper is available in the Hugging Face [Transformers](https://github.com/huggingface/transformers) library from Version 4.23.1, with both PyTorch and TensorFlow implementations. All the official checkpoints can be found on the [Hugging Face Hub](https://huggingface.co/models?search=openai/whisper), alongside [documentation](https://huggingface.co/docs/transformers/model_doc/whisper) and [examples scripts](https://huggingface.co/blog/fine-tune-whisper).

To install transformers and datasets, make sure to run:
```bash
pip install transformers==4.24 datasets==2.6.1 librosa
```
```

File: README.md [hunk 2]
```diff
@@ -129,6 +129,43 @@ result = whisper.decode(model, mel, options)
...
+    batch["prediction"] = processor.tokenizer._normalize(transcription)
+    return batch
+
+result = librispeech_test_clean.map(map_to_pred)
+
+wer = load("wer")
+print(100 * wer.compute(references=result["reference"], predictions=result["prediction"]))
+```
+**Print Output:**
+```python
```
Comment (patrickvonplaten) [inline README.md:165]:
```suggestion
```bash
```

**Comment by ameenba** [comment]
> First of all, thank you for your amazing work on the Whisper project and for open-sourcing the family of pre-trained checkpoints - these are of tremendous benefit to both the open-source/open-science communities!
> 
> Whisper was added to Hugging Face Transformers in v4.23.1 in PyTorch and Tensorflow, making it more accessible to the research community. Since then, usage has been steadily increasing, with model downloads now at ~30k / month. Adding a section on the 🤗 Transformers implementation of Whisper to the README would help highlight how one can evaluate Whisper in ~10 lines of code! We're excited to share with you how the community adopt Whisper for ASR research and production!

this is so cool! Can I ask some questions here around the model you're hosting?

- the real time speech isnt working on the site, is that a limitation of your whisper (just like this whisper) or just the website?
- was this model automatically converted through torchscript/onnx, or was the architecture re-defined in TF so the weights could be transferred?
- have you benchmarked latency of the tensorflow model against the original?
- does the TF model support dynamic inputs or are the inputs a fixed size with padding when needed?
- are there any plans to release a tflite model?

**Comment by sanchit-gandhi** [comment]
Hey @ameenba!

> the real time speech isnt working on the site

We're first going to add chunked ASR, after which we can start to look into real time (https://github.com/huggingface/transformers/issues/19887)

> was this model automatically converted through torchscript/onnx, or was the architecture re-defined in TF

Re-defined, see https://github.com/huggingface/transformers/pull/19378

> have you benchmarked latency of the tensorflow model against the original?

Unsure if this has been done! cc @amyeroberts

> does the TF model support dynamic inputs or are the inputs a fixed size with padding when needed?

Again, cc the TF expert @amyeroberts!

**Comment by jongwook** [comment]
Hi Sanchit and all, as discussed offline, let me close this in favor of a separate post on the Discussions page.
